rclpy/src/rclpy_common/src/common.c

```diff
@@ -83,20 +74,32 @@ rclpy_convert_to_py_qos_policy(void * profile)
   set_result += PyObject_SetAttrString(pyqos_profile, "history", pyqos_history);
   set_result += PyObject_SetAttrString(pyqos_profile, "reliability", pyqos_reliability);
   set_result += PyObject_SetAttrString(pyqos_profile, "durability", pyqos_durability);
+  set_result += PyObject_SetAttrString(pyqos_profile, "lifespan", pyqos_lifespan);
   set_result += PyObject_SetAttrString(pyqos_profile,
       "avoid_ros_namespace_conventions", pyqos_avoid_ros_namespace_conventions);
 
   Py_DECREF(pyqos_depth);
   Py_DECREF(pyqos_history);
   Py_DECREF(pyqos_reliability);
   Py_DECREF(pyqos_durability);
+  Py_DECREF(pyqos_lifespan);
   Py_DECREF(pyqos_avoid_ros_namespace_conventions);
 
   if (0 != set_result) {
     Py_DECREF(pyqos_profile);
     return NULL;
   }
   return pyqos_profile;
+
+fail_qos_setting:
+  Py_DECREF(pyqos_profile);
+  if (pyqos_depth) {Py_DECREF(pyqos_depth);}
```

Comment:
I worry about these because of the goto skipping over the initialization. Depending on the compiler settings these stack variables might not be set to 0 and if they aren't explicitly set to null before one of the goto's gets called then it's arbitrary what they will be.

Reply:
oh - you're right - this pattern won't work as is. You see why I wanted to do it though? Had to manage every single previous value on each check - which was really clunky.

You think it would make sense to initialize all up front, before usage e.g.
```
PyObject * pyqos_depth = nullptr;
PyObject * pyqos_history = nullptr;
...

pyqos_depth = PyLong_FromSize_t(qos_profile->depth);

...
```

Reply:
I do see why you wanted to to do it that way. In general I'm a fan of putting all the local variables at the top of your function and provide an initial value. In this case, doing so would also solve your problem.